src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,38 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            return new ArrayList<>(new BulldozerProducer().get());
+        } else if (type == Excavator.class) {
+            return new ArrayList<>(new ExcavatorProducer().get());
+        } else if (type == Truck.class) {
+            return new ArrayList<>(new TruckProducer().get());
+        }
+        return Collections.emptyList();
```

Comment:
Consider creating a local variable for the machines list before returning it to avoid directly returning the new ArrayList construction within the if-else blocks.